.github/workflows/codeql.yml

```diff
@@ -0,0 +1,78 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ["master"]
+  schedule:
+    - cron: "0 0 * * 1"
+
+permissions:
+  contents: read
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["go", "python"]
```

Comment:
Does this only work for python and go? Then the usefulness of the scan is limited.

Reply:
Ah I see, it's explained at the [Github Guide For Code Scanning](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/setting-up-code-scanning-for-a-repository) link.

Yeah, the usefulness is limited. Perhaps we can execute it only when there are Python code changes?

Reply:
Thanks for using https://github.com/step-security/secure-repo! I am the maintainer of the project. I realized we have a bug due to which `cpp` was not added to the `language` list. CodeQL also supports `cpp`. I have created an [issue](https://github.com/step-security/secure-repo/issues/2016) to address this in the automation. For this PR, can you please add `cpp` to the language list? Or let me know if you want me to update the file using the bot account.

Reply:
I added "cpp"

Reply:
Could we execute this only on C or Python changes? I'm thinking about https://github.com/liblouis/liblouis/pull/1275.

Reply:
Isn't this thing only run once a week due to 
``` yaml
  schedule:
    - cron: "0 0 * * 1"
```
If that is the case then I don't really care if it also runs if there were no code changes

Reply:
Hmm OK then.